module/input/ImageDecompressor/src/decompressor/Decompressor.hpp

```diff
@@ -8,6 +8,13 @@ namespace module::input::decompressor {
 
     class Decompressor {
     public:
+        Decompressor()                        = default;
+        virtual ~Decompressor()               = default;
+        Decompressor(const Decompressor&)     = default;
+        Decompressor(Decompressor&&) noexcept = default;
+        Decompressor& operator=(const Decompressor&) = default;
+        Decompressor& operator=(Decompressor&&) noexcept = default;
```

Comment:
Surely these aren't needed? They are all defaulted

Reply:
I believe they were needed as otherwise the subclasses were getting confused? I wouldn't have put them up here unless clang-tidy yelled at me...